src/printer.js

```diff
@@ -3260,6 +3273,84 @@ function hasLeadingOwnLineComment(text, node) {
   return res;
 }
 
+function hasNakedLeftSide(node) {
+  return (
+    node.type === "BinaryExpression" ||
```

Comment:
Idea: instead of the many many many === statements, what if you were to move all the constants you're testing out into a global const Set? The time to check a set is very performant and this change would make the code smaller, less repetitive and easier to read. If you give the set a nice, easy to understand name, it'll be even better! 

Since this is a function that runs many times, it's worth measuring the actual difference that makes in performance. There's going to have to be a trade off here between performance and readability. I'm suspicious of how the readable version will perform.